website/content/docs/concepts/plugins/storage/host-volumes.mdx

```diff
@@ -0,0 +1,387 @@
+  fi
+  if [ -z "$DHV_VOLUME_NAME" ]; then
+    stderr "DHV_VOLUME_NAME must not be empty"
+    exit 1
+  fi
+  echo "$DHV_HOST_PATH/$DHV_VOLUME_NAME"
+}
+
+is_mounted() {
+  mount | grep -q " $1 "
+}
+
+create_volume() {
+    local path="$1"
+    local bytes="$2"
+
+    # translate to mb for dd block size
+    local megs=$((bytes / 1024 / 1024)) # lazy, approximate
+    if [ $megs -le 0 ]; then
+      stderr "minimum capacity must be greater than zero."
+      exit 2
+    fi
+
+    mkdir -p "$path"
+    # the if statements ensure idempotency
+    if [ ! -f "$path.ext4" ]; then
+      # dd only writes to stderr, so safe to run without redirection
+      dd if=/dev/zero of="$path.ext4" bs=1M count="$megs"
+      # mkfs includes stdout, so we need to redirect that to stderr
+      mkfs.ext4 "$path.ext4" 1>&2
+    fi
+    if ! is_mounted "$path"; then
+      mount "$path.ext4" "$path"
+    fi
+}
+
+delete_volume() {
+  local path="$1"
+  is_mounted "$path" && umount "$path"
+  rm -rf "$path"
+  rm -f "$path.ext4"
+}
+
+case "$1" in
+  "fingerprint")
+    printf '{"version": "%s"}' "$version"
+    ;;
+  "create")
+    path="$(get_path)"
+    stderr "creating volume at $path"
+    create_volume "$path" "$DHV_CAPACITY_MIN_BYTES"
+    # output what Nomad expects
+    bytes="$(stat --format='%s' "$path.ext4")"
+    printf '{"path": "%s", "bytes": %s}' "$path" "$bytes"
+    ;;
+  "delete")
+    path="$(get_path)"
+    stderr "deleting volume at $path"
+    delete_volume "$path" ;;
+  *)
+    echo "unknown operation: $1"
+    exit 1 ;;
+esac
+```
+
+</Tab>
+</Tabs>
+
+## Specification
+
+A plugin is registered with Nomad if it:
+
+* Is an executable file (a script or binary)
+* Is located in the [`client.host_volume_plugin_dir`][plugin_dir] directory
+* Responds appropriately to a `fingerprint` call
+
+### Operations
+
+To fully manage the lifecycle of a volume, plugins must fulfill all of the
+following operations:
+
+* [fingerprint](#fingerprint)
+* [create](#create)
+* [delete](#delete)
+
+Nomad passes the operation as the first positional argument to the plugin.
+That and other information are passed as environment variables. Environment
+variables are prefixed with `"DHV_"` (i.e. Dynamic Host Volume).
+
+Some variables may be required for certain plugins (namely, `DHV_CAPACITY_*`
+for plugins that can restrict size). Most are for plugin author convenience.
+
+#### fingerprint
+<blockquote style={{borderLeft: "solid 1px #00ca8e"}}>
+
+Nomad calls `fingerprint` when the client agent starts (or is reloaded with a
+SIGHUP) to discover valid plugins. The returned "version" is used to register
+the plugin on the Nomad node for volume scheduling.
+
+**CLI arguments:** `$1=fingerprint`
+
+**Environment variables:**
+
+```
+DHV_OPERATION=fingerprint
+```
+
+**Expected stdout:**
+
+```
+{"version": "0.0.1"}
+```
+
+**Requirements:**
+
+* Must complete within 5 seconds, or Nomad will kill it
+  * should be much faster, as no actual work should be done
```

Comment:
Maybe this sentence fragment should be part of the bullet above? (Or in the "considerations" section?)